### PR TITLE
feat: navigation

### PR DESCRIPTION
**Pull request by drinchev**

### CODE REVIEW

File: src/navigation/SecondarySidebar/SecondarySidebar.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import { ActionButton } from '../../components/ActionButton';
+
+/**
+ * Secondary Sidebar is used when you want to render another sidebar next to
+ * the main [Sidebar](#/Navigation/Sidebar) component.
+ *
+ * It is rendered as fixed positioned `aside` html element and passes the
+ * corresponding `props` of the SecondarySidebar component.
+ *
+ * The Secondary Sidebar has optional header that can be used, as well as
```
Comment (blumendorf) [inline src/navigation/SecondarySidebar/SecondarySidebar.js]:
```suggestion
 * The Secondary Sidebar has an optional header that can be used, as well as
```

File: src/navigation/SecondarySidebar/SecondarySidebar.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      <SidebarNavigation>
+        <SidebarNavigationItem icon="home" title="Home" to="/home" />
+        <SidebarNavigationItem icon="sensor" title="Devices" to="/devices" />
+      </SidebarNavigation>
+    </MemoryRouter>
+  </Sidebar>
+  <SecondarySidebar
+    style={{ position: 'absolute' }}
+    header="Filters"
+    open={open}
```
Comment (blumendorf) [inline src/navigation/SecondarySidebar/SecondarySidebar.md]:
```suggestion
    open={open}
    onClickMobileBack={() => setOpen(false)}
```

File: src/navigation/SecondarySidebar/SecondarySidebar.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  /**
+   * Optional header of the sidebar that is rendered at the top
+   */
+  header: node,
+  /**
+   * Controls if the secondary sidebar should be visible
+   */
+  open: bool,
+  /**
+   * Controls if the secondary sidebar should be hidden on mobile devices
```
Comment (blumendorf) [inline src/navigation/SecondarySidebar/SecondarySidebar.js]: This prop is not entirely clear to me. Maybe a short explanation why to do this?

File: src/navigation/Sidebar/Sidebar.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { func, node, string, bool, shape } from 'prop-types';
+import classnames from 'classnames';
+import * as olt from '@lightelligence/styles';
+import { ActionButton } from '../../components/ActionButton';
+
+/**
+ * Sidebar component is used to render a sidebar on the left side of the user's
```
Comment (blumendorf) [inline src/navigation/Sidebar/Sidebar.js]:
```suggestion
 * The Sidebar component is used to render a sidebar on the left side of the user's
```

File: src/navigation/Sidebar/Sidebar.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ * browser.
+ *
+ * It consists of body, which is rendered on the top side of the sidebar and
+ * additional `bottom` property which is rendered on the bottom side of the
```
Comment (blumendorf) [inline src/navigation/Sidebar/Sidebar.js]:
```suggestion
 * an additional `bottom` property which is rendered on the bottom side of the
```

File: src/navigation/SidebarNavigation/SidebarNavigation.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { oneOf, arrayOf, shape, oneOfType } from 'prop-types';
+import React from 'react';
+import { SidebarNavigationItem } from './SidebarNavigationItem';
+
+/**
+ * Sidebar navigation is the navigation for the middle of the sidebar. It is
+ * consisted of [SidebarNavigationItem](#/Navigation/SidebarNavigationItem) and
```
Comment (blumendorf) [inline src/navigation/SidebarNavigation/SidebarNavigation.js]:
```suggestion
 * consists of [SidebarNavigationItem](#/Navigation/SidebarNavigationItem) and
```

File: src/navigation/SidebarNavigation/SidebarNavigation.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { oneOf, arrayOf, shape, oneOfType } from 'prop-types';
+import React from 'react';
+import { SidebarNavigationItem } from './SidebarNavigationItem';
+
+/**
+ * Sidebar navigation is the navigation for the middle of the sidebar. It is
```
Comment (blumendorf) [inline src/navigation/SidebarNavigation/SidebarNavigation.js]:
```suggestion
 * Sidebar navigation is the navigation for the middle of the sidebar. It
```

File: src/navigation/SidebarNavigation/SidebarNavigationItem.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+import { SidebarSubNavigationItem } from './SidebarSubNavigationItem';
+
+/**
+ * Navigation item for the [SidebarNavigation](#/Navigation/SidebarNavigation)
+ * Component.
+ *
+ * The component also passes all other `props` to the underlying
+ * [Link](#/Content/Link) component.
+ *
+ * It always have a rendered icon on the left side of the navigation item.
```
Comment (blumendorf) [inline src/navigation/SidebarNavigation/SidebarNavigationItem.js]:
```suggestion
 * It always displays an icon on the left side of the navigation item.
```

File: src/navigation/SidebarNavigation/SidebarNavigationItem.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+ *
+ * In case of provided children the navigation item renders additional `nav`
```
Comment (blumendorf) [inline src/navigation/SidebarNavigation/SidebarNavigationItem.js]:
```suggestion
 * If children are provided the navigation item renders additional `nav`
```

File: src/navigation/SidebarSelector/SidebarSelectorFilter.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+import classnames from 'classnames';
+import * as olt from '@lightelligence/styles';
+import { SidebarSelectorFilterItem } from './SidebarSelectorFilterItem';
+
+/**
+ * Sidebar selector is a selector component that is used in the Sidebar to
+ * show a filter button.
+ *
+ * You can wire this to a [SecondarySidebar](#/Navigation/SecondarySidebar)
+ * component. Please check [Layout](#/Navigation/Layout) to check for
```
Comment (blumendorf) [inline src/navigation/SidebarSelector/SidebarSelectorFilter.js]:
```suggestion
 * component. Please check [Layout](#/Navigation/Layout) for
```

File: src/navigation/SidebarSelector/SidebarSelectorProperty.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+import React from 'react';
+import classnames from 'classnames';
+import * as olt from '@lightelligence/styles';
+
+/**
+ * Sidebar Property selector is a selector component that is used in the
+ * Sidebar to select a property
+ *
+ * You can wire this to a [SecondarySidebar](#/Navigation/SecondarySidebar)
+ * component. Please check [Layout](#/Navigation/Layout) to check for
```
Comment (blumendorf) [inline src/navigation/SidebarSelector/SidebarSelectorProperty.js]:
```suggestion
 * component. Please check [Layout](#/Navigation/Layout) for
```

File: src/navigation/SidebarSelector/SidebarSelectorTenant.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+import React from 'react';
+import classnames from 'classnames';
+import * as olt from '@lightelligence/styles';
+
+/**
+ * Sidebar selector is a selector component that is used in the Sidebar to
+ * show a tenant switch
+ *
+ * You can wire this to a [SecondarySidebar](#/Navigation/SecondarySidebar)
+ * component. Please check [Layout](#/Navigation/Layout) to check for
```
Comment (blumendorf) [inline src/navigation/SidebarSelector/SidebarSelectorTenant.js]:
```suggestion
 * component. Please check [Layout](#/Navigation/Layout) for
```

File: src/navigation/Layout/Layout.js [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * It gives you a wrapper that will properly position :
+ *
+ * - [Header Component](#/Navigation/Header)
+ * - [Sidebar Component](#/Navigation/Sidebar)
+ * - [SecondarySidebar Component](#/Navigation/SecondarySidebar)
+ * - [LayoutBody Component](#/Navigation/LayoutBody)
+ *
+ * The layout uses a `div` HTML tag and passes all props to it's corresponding
+ * React Element.
```
[the remark this answers is not in the document]
Comment (drinchev) [reply src/navigation/Layout/Layout.js]: Yeah I guess no need for that in the Layout component. Will fix it.

File: src/navigation/SecondarySidebar/SecondarySidebar.js (hunk 3 above)
Comment (drinchev) [reply src/navigation/SecondarySidebar/SecondarySidebar.js]: I will add :)

**Pull request merged**